test/integration/PermissionTable.test.ts

```diff
@@ -0,0 +1,218 @@
+  [ 'PUT',     'C/R', [],                     [ AM.write ],           '',     TXT, 205, 401 ],
+  [ 'PUT',     'C/R', [ AM.read ],            undefined,              '',     TXT, 401, 401 ],
+  [ 'PUT',     'C/R', [ AM.append ],          undefined,              '',     TXT, 401, 401 ],
+  [ 'PUT',     'C/R', [ AM.write ],           undefined,              '',     TXT, 205, 201 ],
+  [ 'PUT',     'C/R', [ AM.append ],          [ AM.write ],           '',     TXT, 205, 201 ],
+
+  [ 'PATCH',  'C/R', [],                     undefined,              DELETE,  N3,  401, 401 ],
+  // We don't return 404 yet in case a PATCH has no inserts and C/R does not exist
+  // [ 'PATCH',   'C/R', [],                     [ AM.read ],            DELETE,  N3,  401, 404 ],
+  [ 'PATCH',   'C/R', [],                     [ AM.append ],          INSERT, N3,  205, 401 ],
+  [ 'PATCH',   'C/R', [],                     [ AM.append ],          DELETE, N3,  401, 401 ],
+  [ 'PATCH',   'C/R', [],                     [ AM.write ],           INSERT, N3,  205, 401 ],
+  [ 'PATCH',   'C/R', [],                     [ AM.write ],           DELETE, N3,  401, 401 ],
+  [ 'PATCH',   'C/R', [ AM.append ],          [ AM.write ],           INSERT, N3,  205, 201 ],
+  [ 'PATCH',   'C/R', [ AM.append ],          [ AM.write ],           DELETE, N3,  401, 401 ],
+  // We don't return 404 yet in case a PATCH has no inserts and C/R does not exist
+  // [ 'PATCH',   'C/R', [],                     [ AM.read, AM.write ],  DELETE, N3,  205, 404 ],
+
+  [ 'DELETE',  'C/R', [],                     undefined,              '',     '',  401, 401 ],
+  [ 'DELETE',  'C/R', [],                     [ AM.read ],            '',     '',  401, 404 ],
+  [ 'DELETE',  'C/R', [],                     [ AM.append ],          '',     '',  401, 401 ],
+  [ 'DELETE',  'C/R', [],                     [ AM.write ],           '',     '',  401, 401 ],
+  [ 'DELETE',  'C/R', [ AM.read ],            undefined,              '',     '',  401, 404 ],
+  [ 'DELETE',  'C/R', [ AM.append ],          undefined,              '',     '',  401, 401 ],
+  [ 'DELETE',  'C/R', [ AM.append ],          [ AM.read ],            '',     '',  401, 404 ],
+  // Not sure why 401 is suggested here instead of 404?
+  // [ 'DELETE',  'C/R', [ AM.write ],           undefined,              '',     '',  205, 401 ],
+  [ 'DELETE',  'C/R', [ AM.write ],           [ AM.read ],            '',     '',  401, 404 ],
+  [ 'DELETE',  'C/R', [ AM.write ],           [ AM.append ],          '',     '',  401, 401 ],
+
+  [ 'DELETE',  'C/',  [],                     undefined,              '',     '',  401, 401 ],
+  [ 'DELETE',  'C/',  [ AM.read ],            undefined,              '',     '',  401, 404 ],
+  [ 'DELETE',  'C/',  [ AM.append ],          undefined,              '',     '',  401, 401 ],
+  // You need read permissions to delete a container?
+  // [ 'DELETE',  'C/',  [ AM.write ],           undefined,              '',     '',  401, 401 ],
```

Comment:
This one might need it since I couldn't find anything saying you need read permissions to delete a container. It might make sense since you can only delete empty containers so this implies a read, but this is not strictly written in a spec anywhere that I could find.

Reply:
Container deletion requires that container to be empty. If no Read permissions, you would disclose whether a container is empty by deleting it.